src/backend/distributed/commands/multi_copy.c

```diff
@@ -610,7 +839,16 @@ CopyToNewShards(CopyStmt *copyStatement, char *completionTag)
 			{
 				SendCopyBinaryFooters(copyOutState, shardConnections->connectionList);
 				FinalizeCopyToNewShard(shardConnections);
-				UpdateShardStatistics(relationId, shardConnections->shardId);
+
+				if (masterConnection == NULL)
+				{
+					UpdateShardStatistics(relationId, shardConnections->shardId);
+				}
+				else
+				{
+					UpdateRemoteShardStatistics(masterConnection, relationName,
```

Comment:
> It would be kind be nice if we computed shardlength, shardminvalue, shardmaxvalue during the COPY so we could avoid this 3-way interaction and instead just send an UPDATE to the master.

One problem with the shardlength is that we can't actually calculate the exact shard size from the data we send. This is more obvious for cstore_fdw cases.